mantidproject/mantid
# Add C++ version of SortXAxis #22515

**Description of work.**
It was requested that for C++ users the algorithm SortXAxis was rewritten in C++ so it is more easily used, in a C++ context instead of writing it individually across. This also replaces the Python as the C++ is V2 and the Python is V1.

**Report to:** nobody  

**To test:**

- Either Load some data can be anything but ` TestSampleData->D33041421_tof.nxs ` is a good one to use and run the algorithm like you would any other Algorithm. It should use the C++ variant, which should be inherently faster than Python anyway, which with test data sets as large as this is noticeable.
or
- Run the ctest 'SortXAxisTest.py' 



Fixes #22515 

Does this update require release notes?


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
